drift/lib/src/web/sql_js.dart

```diff
@@ -134,7 +137,15 @@ class SqlJsDatabase {
       // matches the behavior from a `NativeDatabase`.
       _obj.run(sql, _undefined);
     } else {
-      _obj.run(sql, args);
+      _obj.run(
+          sql,
+          args.map((e) {
+            if (e != null && e is BigInt) {
+              return _bigInt(e.checkRange.toString());
+            } else {
+              return e;
+            }
+          }).toList());
```

Comment:
Let's use a collection for here, something like

```dart
[for (final arg in args) if (e is BigInt) _bigInt(e.checkRange.toString()) else e]
```

(Also no need for the `e != null`, it's included in the `is` even without null-safety).